src/components/Discover/Discover.js

```diff
@@ -1,7 +1,45 @@
 import './Discover.css';
+import React, { useState, useEffect, useContext } from "react"
+import UserContext from '../UserContext/UserContext';
+import { getAlbumsByMasterId, getAlbumsByGenre } from '../../api'
+import RecommendedAlbum from '../RecommendedAlbum/RecommendedAlbum';
 
-const Discover = () => {
-  return <h2>Yours to Discover</h2>
+
+const Discover = ({ trendingData }) => {
+  const { currentUser } = useContext(UserContext)
+  const [ recommendedData, setRecommendedData ] = useState()
+
+  useEffect(() => {
+    if (!currentUser || !currentUser.collections.length) {
+      setRecommendedData(trendingData)
+    } else {
+      const randomAlbum = currentUser.collections[Math.floor(Math.random() * currentUser.collections.length)]
+      getAlbumsByMasterId(randomAlbum.masterId)
+      .then(data => getAlbumsByGenre(data.genres[0]))
+      .then(data => {
+        let recommendedAlbumsData = [];
+        for (let i = 0; recommendedAlbumsData.length < 10; i++) {
```

Comment:
this exit condition could be an issue if any genre has less than 10 entries... thoughts on how to handle this case?

Reply:
possibly  check if there are enough entries available in the genre before entering the loop.Check if there are at least 10 albums available in the genre? Then handle the case when there are not enough albums in the genre?

Reply:
I had a similar problem in What's Cookin'.  I think the database here is so big that it won't be an issue, but the solution I found was to introduce some recursion, where it would call the function with a random category until it the array i wanted had the desired length.  Wouldn't necessarily solve the problem if you're always using the first item in data.genres though

Reply:
we can make a conditional where if the length of the fetch array is less than 10 then limit it to the number in the array. We can save this as a number in a variable and put that in the for loop

Reply:
I just did the minimum and added the check that sets the max to the results length if less than 10
`const maxResults = data.results.length < 10 ? data.results.length : 10`

Reply:
@jalbe0076 great minds!